owltorrent-core/src/main/java/edu/rice/owltorrent/core/serialization/TempParser.java

```diff
@@ -2,19 +2,104 @@
 
 import com.dampcake.bencode.BencodeInputStream;
 import edu.rice.owltorrent.common.entity.Torrent;
+import lombok.NonNull;
+
 import java.io.ByteArrayInputStream;
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 
+/**
+ * Torrent file parser.
+ *
+ * @author bhaveshshah
+ */
 public class TempParser {
 
-  public Torrent parse(File file) throws IOException {
-    // TODO: revise byte reading method
+  /**
+   * Bencode a Torrent file and extract important attributes
+   *
+   * @param file Torrent file path
+   * @return pruned Torrent object
+   */
+  public static Torrent parse(@NonNull File file) throws IOException {
+    var dict = bencode(file);
+    Torrent torrent = extractAttributes(dict);
+    return torrent;
+  }
+
+  /**
+   * Bencode a Torrent file
+   *
+   * @param file Torrent file
+   * @return dictionary containing all file attributes
+   */
+  public static Map<String, Object> bencode(@NonNull File file) throws IOException {
     ByteArrayInputStream inputStream = new ByteArrayInputStream(Files.readAllBytes(file.toPath()));
     BencodeInputStream bencodeInputStream = new BencodeInputStream(inputStream);
     var dict = bencodeInputStream.readDictionary();
-    System.out.println(dict.keySet());
-    return new Torrent();
+
+    return dict;
+  }
+
+  /**
+   * Extract important attributes to store for later use
+   *
+   * @param dict Containing all file attributes
+   * @return pruned Torrent object
+   */
+  public static Torrent extractAttributes(@NonNull Map<String, Object> dict) {
+    var infoDict = (Map<String, Object>) dict.get("info");
+
+    String announceURL = (String) dict.get("announce");
+    String name = (String) infoDict.get("name");
+    long pieceLength = (long) infoDict.get("piece length");
+    List<String> pieces = extractPieces((String) infoDict.get("pieces"));
+    long length = getFileLength((ArrayList<LinkedHashMap<String, String>>) infoDict.get("files"));
```

Comment:
For multi File torrents, do we want to calculate a total length? I think would want lengths for each file?
Also I don't think this code works for single file torrents, since they won't have a files field?

checkout: https://wiki.theory.org/BitTorrentSpecification#Info_Dictionary

Reply:
Stored lengths + paths for each file instead. Also modified code to support single file torrents as well.